statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np 
+import pandas as pd
+#from numbers import Number
+
+
+class SurveyStat(object):
+
+    def __init__(self, data, cluster=None, strata=None, prob_weights=None):
+
+        data, self.cluster = self._check_type(data, cluster)
+        data, self.strata = self._check_type(data, strata)
+        data, self.prob_weights = self._check_type(data, prob_weights)
+        
+        n = data.shape[0]
+        p = data.shape[1]
+        self.data = np.asarray(data).reshape(n,p)
+
+    # preprocessing to be used by init
+    def _check_type(self, data, sampling_method):
+        """
+        converts cluster, strata, etc to an numpy array
+        gets rid of column from data if possible
+
+        Parameters
+        ----------
+        data : array-like
+          The raw data
+        sampling_method : one of cluster, strata, or prob_weight
+
+        Returns
+        -------
+        A nx1 numpy array
+        """
+
+        # check if str, num, or list
+        if isinstance(sampling_method, str):
+            temp = data[sampling_method].values
+            del data[sampling_method]
+            sampling_method = temp
+            n = len(sampling_method)
+            sampling_method = sampling_method.reshape(n,1)
+        elif isinstance(sampling_method, int):
+            try:
+                temp = data.iloc[:, sampling_method].values
+                data.drop(data.columns[sampling_method], inplace=True, axis=1)
+                sampling_method = temp
+                n = len(sampling_method)
+                sampling_method = sampling_method.reshape(n,1)
+            except AttributeError: # data is a ndarray
+                temp = data[:,sampling_method]
+                data = np.delete(data, sampling_method, 1)
+                sampling_method = temp
+                n = len(sampling_method)
+                sampling_method = sampling_method.reshape(n,1)
+        elif isinstance(sampling_method, list):
+            sampling_method = np.array(sampling_method)
+        elif sampling_method is None:
+            n = data.shape[0]
+            self.sampling_method = np.ones([n,1])
+        else:
+            n = len(sampling_method)
+            sampling_method = sampling_method.reshape(n,1)
+
+        return data, sampling_method
+
+
+    def _create_subgroup_labels(self):
+        """
+        creates labels for each SSU within a PSU so that one is sure that 
+        operations are being done withing each sampling subgroup
+
+        Parameters
+        ----------
+        self
+
+        Returns
+        -------
+        A nx3 array, the first column being self.strata, the second column
+        is self.cluster, the third column is the unique label for each subgroup
+        """
+
+        _, i1 = np.unique(strata, return_inverse=True)
+        _, i2 = np.unique(clusters, return_inverse=True)
+        labels = i1 * len(i2) + i2
+        return labels
+
+    def _jackknife(self, method, column_index):
```

Comment:
Some of this function is book-keeping (getting the indices for strata and clusters) which will be needed for all methods, not just jack-knife.   That code is mixed in here with the jack-knife itself.  I would separate the index book-keeping as a separate function and have the jackknife function only do the work that is specific to jackknife.